packages/autocomplete-generators/src/keyvalue.ts

```diff
@@ -0,0 +1,79 @@
+export interface KeyValueInit {
+  key:
+    | Fig.Suggestion[]
+    | ((
+        tokens: string[],
+        executeShellCommand: Fig.ExecuteShellCommandFunction
+      ) => Promise<Fig.Suggestion[]>);
+  sep: string;
+  value:
+    | Fig.Suggestion[]
+    | ((
+        tokens: string[],
+        executeShellCommand: Fig.ExecuteShellCommandFunction
+      ) => Promise<Fig.Suggestion[]>);
+}
+
+export interface KeyValueListInit extends KeyValueInit {
+  delim: string;
+}
+
+export function keyvalue({ key, sep, value }: KeyValueInit): Fig.Generator {
+  return {
+    trigger: (newToken, oldToken) => newToken.indexOf(sep) !== oldToken?.indexOf(sep),
+    getQueryTerm: (token) => (token as string).slice((token as string).indexOf(sep) + 1),
+    custom: async (tokens, executeShellCommand) => {
+      const finalToken = tokens[tokens.length - 1];
+      const isKey = !finalToken.includes(sep);
+
+      if (isKey) {
+        if (typeof key === "function") {
+          return key(tokens, executeShellCommand);
+        }
+        return key;
+      }
+
+      if (typeof value === "function") {
+        return value(tokens, executeShellCommand);
+      }
+      return value;
+    },
+  };
+}
+
+function getFinalSepDelimIndex(sep: string, delim: string, token: string): number {
+  const sepIdx = token.lastIndexOf(sep);
+  const delimIdx = token.lastIndexOf(delim);
+  return Math.max(sepIdx, delimIdx);
+}
+
+export function keyvaluelist({ key, sep, value, delim }: KeyValueListInit): Fig.Generator {
```

Comment:
super nitpicking but I would keep the camel case for generators too
```suggestion
export function keyValueList({ key, sep, value, delim }: KeyValueListInit): Fig.Generator {
```

Reply:
Only reason it wasn't is because of `filepaths` - I really don't care how this is cased so happy to change it 😄

Reply:
I think filepath is commonly cased as a single word. At least in node😁